cli/azd/pkg/azapi/stack_deployments.go

```diff
@@ -0,0 +1,628 @@
+				ResourceGroups:   &deleteBehavior,
+			},
+			DenySettings: &armdeploymentstacks.DenySettings{
+				Mode: to.Ptr(armdeploymentstacks.DenySettingsModeNone),
+			},
+			Parameters: stackParams,
+			Template:   armTemplate,
+		},
+	}
+	poller, err := client.BeginCreateOrUpdateAtSubscription(ctx, deploymentName, stack, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = poller.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return d.GetSubscriptionDeployment(ctx, subscriptionId, deploymentName)
+}
+
+func (d *StackDeployments) DeployToResourceGroup(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroup string,
+	deploymentName string,
+	armTemplate azure.RawArmTemplate,
+	parameters azure.ArmParameters,
+	tags map[string]*string,
+) (*ResourceDeployment, error) {
+	client, err := d.createClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	stackParams := map[string]*armdeploymentstacks.DeploymentParameter{}
+	for k, v := range parameters {
+		stackParams[k] = &armdeploymentstacks.DeploymentParameter{
+			Value: v.Value,
+		}
+	}
+
+	deleteBehavior := armdeploymentstacks.DeploymentStacksDeleteDetachEnumDelete
+
+	stack := armdeploymentstacks.DeploymentStack{
+		Tags: tags,
+		Properties: &armdeploymentstacks.DeploymentStackProperties{
+			BypassStackOutOfSyncError: to.Ptr(false),
+			ActionOnUnmanage: &armdeploymentstacks.ActionOnUnmanage{
+				Resources:        &deleteBehavior,
+				ManagementGroups: &deleteBehavior,
+				ResourceGroups:   &deleteBehavior,
+			},
+			DenySettings: &armdeploymentstacks.DenySettings{
+				Mode: to.Ptr(armdeploymentstacks.DenySettingsModeNone),
+			},
+			Parameters: stackParams,
+			Template:   armTemplate,
+		},
+	}
+	poller, err := client.BeginCreateOrUpdateAtResourceGroup(ctx, resourceGroup, deploymentName, stack, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = poller.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return d.GetResourceGroupDeployment(ctx, subscriptionId, resourceGroup, deploymentName)
+}
+
+func (d *StackDeployments) ListSubscriptionDeploymentOperations(
+	ctx context.Context,
+	subscriptionId string,
+	deploymentName string,
+) ([]*armresources.DeploymentOperation, error) {
+	deployment, err := d.GetSubscriptionDeployment(ctx, subscriptionId, deploymentName)
+	if err != nil && !errors.Is(err, ErrDeploymentNotFound) {
+		return nil, err
+	}
+
+	if deployment != nil && deployment.DeploymentId != "" {
+		deploymentName = filepath.Base(deployment.DeploymentId)
+	}
+
+	return d.standardDeployments.ListSubscriptionDeploymentOperations(ctx, subscriptionId, deploymentName)
+}
+
+func (d *StackDeployments) ListResourceGroupDeploymentOperations(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroupName string,
+	deploymentName string,
+) ([]*armresources.DeploymentOperation, error) {
+	// The requested deployment name may be an inner deployment which will not be found in the deployment stacks.
```

Comment:
Example:

- We are using subscription level deployment stack
- The deployment included nested deployments
- Other components (AzureResourceManager) are iterating over deployment operations and find a nested deployment object.  
- This method is called to get the operations of the nested resource group deployment.  
- It is not a deployment stack at this point and the get will fail with 404... so we revert to call the base standard deployment to get the deployment operations of the RG deployment.

Reply:
I guess the thing that feels odd to me here is that AzureResourceManager knows when it finds a nested deployment object that to get the operations from it, it knows that it is a resource group scoped deployment and so it should never try to do anything related to a deployment stack.  I feel like the reason this code is here is because how we ended up factoring logic across interfaces and we are mixing different types of data together.  I'm wondering what would happen if there was a deployment stack with the same name as this deployment object, would we get confused?

Reply:
I think in this use case you are correct.  We don't need to bring in the dynamic deployment service at this point since it should always use the standard deployments API when looking up nested deployments.  I will make this change and test before getting this merged in.